modules/store/src/reducer_creator.ts

```diff
@@ -0,0 +1,59 @@
+import { ActionCreator, ActionType, Action } from './models';
+
+export type On<S> = (state: S, action: Action) => S;
+export type OnReducer<S, C extends ActionCreator[]> = (
+  state: S,
+  action: ActionType<C[number]>
+) => S;
+export type Reducer<S> = (state: S | undefined, action: Action) => S;
+
+export function on<C1 extends ActionCreator, S>(
+  creator1: C1,
+  reducer: OnReducer<S, [C1]>
+): { reducer: On<S>; types: string[] };
```

Comment:
Should we create a type for `: { reducer: On<S>; types: string[] };` ? We're using it several times here

Reply:
I moved things a bit there... should be better now.

Reply:
I moved things around a bit